rskj-core/src/test/java/co/rsk/peg/feeperkb/FeePerKbSupportImplTest.java

```diff
@@ -0,0 +1,221 @@
+package co.rsk.peg.feeperkb;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import co.rsk.bitcoinj.core.Coin;
+import co.rsk.peg.BridgeSerializationUtils;
+import co.rsk.peg.feeperkb.constants.FeePerKbConstants;
+import co.rsk.peg.vote.ABICallElection;
+import co.rsk.peg.vote.ABICallSpec;
+import co.rsk.peg.vote.AddressBasedAuthorizer;
+import java.util.Objects;
+import java.util.Optional;
+import org.ethereum.core.BlockTxSignatureCache;
+import org.ethereum.core.ReceivedTxSignatureCache;
+import org.ethereum.core.SignatureCache;
+import org.ethereum.core.Transaction;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class FeePerKbSupportImplTest {
+
+    @Mock
+    FeePerKbStorageProvider provider;
+    @Mock
+    FeePerKbConstants feePerKbConstants;
+    @InjectMocks
+    FeePerKbSupportImpl feePerKbSupport;
+
+
+    @Test
+    void getFeePerKb() {
+        Coin mockCoin = mock(Coin.class);
+        Optional<Coin> currentFeePerKb = Optional.of(mockCoin);
+        when(provider.getFeePerKb()).thenReturn(currentFeePerKb);
+
+        assertEquals(feePerKbSupport.getFeePerKb(), currentFeePerKb.get());
+    }
+
+    @Test
+    void voteFeePerKbChange_unsuccessfulVote_unauthorized() {
+        AddressBasedAuthorizer authorizer = mock(AddressBasedAuthorizer.class);
+        when(feePerKbConstants.getFeePerKbChangeAuthorizer()).thenReturn(authorizer);
+
+        Transaction tx = mock(Transaction.class);
+        SignatureCache signatureCache = spy(
+            new BlockTxSignatureCache(new ReceivedTxSignatureCache()));
+        when(authorizer.isAuthorized(tx, signatureCache)).thenReturn(false);
+
+        assertEquals(Integer.valueOf(FeePerKbResponseCode.UNAUTHORIZED_CALLER.getCode()),
```

Comment:
Also here, split the values obtained from the comparison

Reply:
Done!